lib/init.coffee

```diff
@@ -18,7 +18,7 @@ module.exports =
 
   provideLinter: ->
     provider =
-      grammarScopes: ['source.haskell']
+      grammarScopes: ['source.haskell', 'meta.embedded.haskell', 'text.tex.latex.haskell']
```

Comment:
I don't think the `meta.embedded.haskell` should be included, as that could be used for _any_ embedded Haskell. `text.tex.latex.haskell` was present in both of your examples and should be sufficient.